src/focus-history.js

```diff
@@ -43,7 +43,8 @@ export class FocusHistory {
 
     /** @private @const {function(!Event)} */
     this.captureFocus_ = e => {
-      if (e.target) {
+      // Hack (#15079) due to Firefox firing focus events on the entire page
+      if (e.target && e.target.nodeType == 1) {
```

Comment:
Firefox doesn't seem to fire any focus event with an element as the target.

Reply:
But it does... It fires normal focus events where the element is a target. It just also happens to capture extra focus events where the target is the entire page, in which case it's an HTML Document, and not an Element. Is there a better way to resolve this problem or have I misdiagnosed it?

Reply:
How do I get it to fire on an element from your test page?

Reply:
Tab through the page, or click on the carousel arrow buttons.

Reply:
<img width="1058" alt="screen shot 2018-05-04 at 2 23 49 pm" src="https://user-images.githubusercontent.com/1528181/39653685-d769514c-4fa6-11e8-95cb-348198e9ffd1.png">
For example, you can run this on localhost, and just add a console log to the `captureFocus_` function. If you click on the hero image to trigger a lightbox open, it will run `captureFocus_` on the document. If you click on the amp-carousel's arrows, which are focusable, it will trigger `captureFocus_` on the arrow button element. 

If you tab through the document, it will trigger a strange series of `captureFocus_` calls on the document and on various focusable elements on the page.